PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azure/toolkit/eclipse/common/component/AzWizardPageWrapper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.eclipse.common.component;
+
+import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
+import com.microsoft.azure.toolkit.lib.common.form.AzureValidationInfo;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import org.eclipse.jface.wizard.IWizardPage;
+import org.eclipse.jface.wizard.WizardPage;
+import reactor.core.Disposable;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public abstract class AzWizardPageWrapper<T> extends WizardPage {
+    private Disposable subscription;
+    private int validationDelay = 300;
+    private volatile boolean validationEnabled = false;
+
+    protected AzWizardPageWrapper(String pageName) {
+        super(pageName);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopValidation();
+        validationEnabled = false;
+    }
+
+    private void startValidation() {
+        stopValidation();
+
+        this.getForm().getInputs().forEach(t -> t.addValueChangedListener(this::valueChanged));
+    }
+
+    private void valueChanged(Object o) {
+        validateAll();
+    }
+
+    private boolean validateAll() {
+        if (!validationEnabled) {
```

Comment:
return true if validation not enabled? Or if there is no need to do validate for a page, it will block users to get into next page.

Reply:
This code is copied from AzDialogWrapper